pyv/transpiler.py

```diff
@@ -0,0 +1,612 @@
+                b.self_type = node.name
+
+        object_def = "type\n"
+        object_def += self.indent(f"{node.name} = object\n", level=node.level + 1)
+        object_def += "\n".join([self.indent(f, level=node.level + 2) for f in fields])
+        body = [self.visit(b) for b in node.body]
+        body = "\n".join(body)
+        return f"{object_def}\n{body}\n"
+
+    def visit_IntEnum(self, node):
+        fields = []
+        for member, var in node.class_assignments.items():
+            var = self.visit(var)
+            if var == "auto()":
+                fields.append(f"{member},")
+            else:
+                fields.append(f"{member} = {var},")
+        fields = "\n".join([self.indent(f) for f in fields])
+        return f"type {node.name} = enum\n{fields}\n\n"
+
+    def visit_IntFlag(self, node):
+        fields = []
+        for member, var in node.class_assignments.items():
+            var = self.visit(var)
+            if var == "auto()":
+                fields.append(f"{member},")
+            else:
+                fields.append(f"{member} = {var},")
+        fields = "\n".join([self.indent(f, level=2) for f in fields])
+        # flags = self.indent(f"{node.name}Flags = set[{node.name}]")
+        return "\n".join(
+            [
+                "type",
+                self.indent(f"{node.name} = enum"),
+                f"{fields}",
+                # f"{flags}",
+                "",
+            ]
+        )
+
+    def visit_StrEnum(self, node):
+        fields = []
+        for member, var in node.class_assignments.items():
+            var = self.visit(var)
+            if var == "auto()":
+                fields.append(f"{member},")
+            else:
+                fields.append(f"{member} = {var},")
+        fields = "\n".join([self.indent(f) for f in fields])
+        return f"type {node.name} = enum\n{fields}\n\n"
+
+    def visit_alias(self, node):
+        return "use {0}".format(node.name)
+
+    def visit_Import(self, node):
+        imports = [self.visit(n) for n in node.names]
+        return "\n".join(i for i in imports if i)
+
+    def visit_ImportFrom(self, node):
+        if (
+            node.module not in self.ALLOW_MODULE_LIST
+            and node.module in self.IGNORED_MODULE_LIST
+        ):
+            return ""
+
+        names = [n.name for n in node.names]
+        names = ", ".join(names)
+        module_path = node.module.replace(".", "::")
+        return f"from {module_path} import {names}"
+
+    def visit_List(self, node):
+        elements = [self.visit(e) for e in node.elts]
+        elements = ", ".join(elements)
+        return f"@[{elements}]"
+
+    def visit_Set(self, node):
+        self._usings.add("sets")
+        elements = [self.visit(e) for e in node.elts]
+        elements_str = ", ".join(elements)
+        return f"toHashSet([{elements_str}])"
+
+    def visit_Dict(self, node):
+        self._usings.add("tables")
```

Comment:
`dict.v` fails with
```
dict.v:1:8: warning: module 'tables' is imported but never used
    1 | import tables
      |        ~~~~~~
    2 |
    3 | fn implicit_keys() bool {
builder error: cannot import module "tables" (not found)
```

Is that a Vlang module which needs to be installed into CI?

Or is this copied from pynim?  It seems a few of these were copied from there.

Syntax which isnt supported yet can raise [`AstNotImplementedError`] found in https://github.com/adsharma/py2many/blob/main/py2many/exceptions.py